### ISSUE TITLE
feat(Serializers): Provide first class STJ support in a box

### ISSUE THREAD

**Issue by bartelink** [Azure/azure-cosmos-dotnet-v3#4325]
TL;DR I and the world want to ref a package that implements a STJ LINQ serializer on a turnkey basis, along the lines suggested in the xmldoc at https://github.com/Azure/azure-cosmos-dotnet-v3/blob/master/Microsoft.Azure.Cosmos/src/Serializer/CosmosLinqSerializer.cs#L17

This would be a fitting conclusion to the long saga of adding this support, which has ended with what seems like a thorough and complete implementation once the work got prioritized. This fit and finish / polish for me is not optional, and should not be left so a thousand flowers can bloom. There will not be flowers the way this is set up atm.

The workaround is for me and hundreds to thousands of other people to pour time into:
1. reading that logic and incorporating it into their many long forgotten implementations of the basic STJ Serialization implementation (such as mine: https://github.com/jet/equinox/blob/master/src/Equinox.CosmosStore/CosmosStoreSerialization.fs#L34-L48)
2. perpetually burning hours on stack overflow and/or traversing issues in this repo wondering whether the impl I have is the latest correct canonical implementation

----

My proposal would be to
1. make CosmosStjSerializer package. Declare it to be preview and/or unsupported, but put a decent placeholder impl in it
2. Make a CosmosNsjSerializer package (it would have a newtonsoft.json ref)
3. make the main SDK depend on CosmosNsjSerializer for now
4. eventually 
  - make a CosmosSerializerless package that is the full SDK, but without default wiring (or refs to) newtonsoft.json
  - make the Cosmos package be a very simple package that refs and binds the CosmosNsjSerializer + CosmosSerializerless packages so legacy usage can just work

Once step 1 is done, the majority of the new world can depend on Cosmos + CosmosStjSerializer and have a DRY one liner config story per repo (I'd do that in Equinox.CosmosStore)

When step 4 is done, I'd adjust my ref to Cosmos to instead be to CosmosSerializerless, and my last reference to newtonsoft.json melts away

_I am well aware this is not trivial to do, and nobody wants to maintain more packages, but the bottom line is that the real world wants and needs to use Cosmos with STJ without having to transplant stuff they don't understand from samples or stackoverflow **for every repo**; An experience where the last piece is an ugly hot potato is not truly solving the whole problem of supporting STJ. This is not because STJ is a MS project, it's because, like it or not, the LINQ support was the only blocker to using Cosmos and STJ in 99% of new codebases_

**Comment by kirankumarkolli** [comment]
AFAIU there are two aspects proposal covers
- STJ implementation
- Netonsoft package dependency (and getting loaded at runtime too)

We can ship STJ implementation part of the SDK it-self. It will help many CX to simply use it with CosmosClientOptions. @kundadebdatta is tracking this work items already. 

Even with CosmosSTJSerializer OOB from Cosmos package, Json.NET will still be loaded at runtime. @bartelink will this address your scenarios?

**Comment by bartelink** [comment]
Yes given the SDK is no lightweight package, a reference to STJ would hardly be a crazy thing to add; until such time as 99% of workloads no longer use STJ (hopefully that's never!)

@kundadebdatta if you have a ref to the issue that tracks this, I'm open to closing this one as long as it conveys the main point - there should be a one-two liner way of asking for STJ instead of NSJ for the 99% case where nobody wants to customize anything (and as a result, any code with behavior in it that goes into an app repository is understood by nobody and just sits there carrying quirks for years to decades into the future)

In the medium term, I think obsoleting the default constructor that opts into NSJ and redirecting via a message to some replacement factory method that forces you to pick NSJ or STJ before eventually removing the NSJ defaulting path and then moving it to an external package is probably for the best.

**Comment by kundadebdatta** [comment]
@bartelink : I have created an issue #4330  in our v3 repo for us to track this work item. Please feel free to close this one.

**Comment by bartelink** [comment]
Thanks @kundadebdatta
